doc/src/guides/solving/solving-guidance.md

```diff
@@ -0,0 +1,221 @@
+>>> for solution in solutions:
+...     print(solution[x].n())
+-0.0509758447494279 + 0.313552108895239*I
+0.0509758447494279 + 3.14751999969868*I
+0.0509758447494279 - 3.14751999969868*I
+-0.0509758447494279 - 0.313552108895239*I
+```
+
+### Equations With no Analytical Solution
+
+The vast majority of arbitrary nonlinear equations are not analytically
+solvable. The classes of equations that are solvable are basically:
+1. Linear equations
+2. Polynomials, except where limited by the Abel-Ruffini theorem
+3. Equations that can be solved by inverting some transcendental functions
+4. Problems that can be transformed into the cases above (e.g., by turning
+trigonometric functions into polynomials)
+5. A few other special cases that can be solved with something like the
+{class}`Lambert W function <sympy.functions.elementary.exponential.LambertW>` 
+
+SymPy may reflect that your equation has no solutions that can be expressed
+algebraically (symbolically) by returning an error such as
+`NotImplementedError`:
+
+```py
+>>> from sympy import solve, cos
+>>> from sympy.abc import x
+>>> solve(cos(x) - x, x, dict=True)
+Traceback (most recent call last):
+  ...
+NotImplementedError: multiple generators [x, cos(x)]
+No algorithms are implemented to solve equation -x + cos(x)
+```
+
+so you may have to {func}`solve your equation numerically
+<sympy.solvers.solvers.nsolve>` instead, for example
+
+```py
+>>> from sympy import nsolve, cos
+>>> from sympy.abc import x
+>>> nsolve(cos(x) - x, x, 2)
+0.739085133215161
+```
+
+If you receive non-closed-form solutions such as {class}`CRootOf
+<sympy.polys.rootoftools.ComplexRootOf>`, you can evaluate them numerically
+using {func}`evalf n <sympy.core.evalf.EvalfMixin.n>`:
```

Comment:
I think Oscar mentioned this, but evalf is also useful even for closed-form solutions (especially when they're large, but even when they're not large, sometimes you just want to know the decimal value of a number).

Reply:
Is this supposed to be `evalf/n`? IMO we should just mention one to avoid confusion (I personally prefer `evalf`).